tests/blackbox.py

```diff
@@ -440,3 +477,412 @@ def dryrun_one(
+            per method X config combination
+        * txn_params: TxParams (optional) - other TxParams to append
+            -in addition to the (is_app_create, OnComplete) information
+        * model_version: int - for compiling self.model_router
+        * model_assemble_constants: bool (optional) - for compiling self.model_router
+        * model_optimize: OptimizeOptions (optional) - for compiling self.model_router
+        * msg: string (optional) - message to report when an assertion is violated
+    """
+
+    def __init__(
+        self,
+        router: Router,
+        predicates: CallPredicates,
+        *,
+        model_router: Router | None = None,
+        algod: v2client.algod.AlgodClient | None = None,
+    ):
+        self.router: Router = self._validate_router(router)
+
+        self.predicates: CallPredicates = self._validate_predicates(predicates)
+
+        self.model_router: Router | None = None
+        if model_router:
+            self.model_router = self._validate_router(model_router, kind="Model")
+
+        self.algod: v2client.algod.AlgodClient = algod or algod_with_assertion()
+
+        self.auto_path_results: Dict[
+            str | None, dict[tuple[bool, OnComplete], SimulationResults]
+        ] = _AutoPathDict()
+
+    # ---- Validation ---- #
+
+    @classmethod
+    def _validate_router(cls, router: Router, kind: str = "Base") -> Router:
+        assert isinstance(router, Router), (
+            f"Wrong type for {kind} Router: {type(router)}. Please provide: " f"Router."
+        )
+        cls._validate_method_configs(router.method_configs)
+
+        return router
+
+    @classmethod
+    def _validate_method_configs(cls, method_configs):
+        assert isinstance(
+            method_configs, (dict)
+        ), f"method_configs '{method_configs}' has type {type(method_configs)} but only 'dict' and 'NoneType' are allowed."
+
+        assert (
+            method_configs
+        ), "make sure to give at least one key/value pair in method_configs"
+
+        for call, meth_config in method_configs.items():
+            assert isinstance(  # type: ignore
+                call, CallType
+            ), f"method_configs dict key '{call}' has type {type(call)} but only str and NoneType are allowed."
+            cls._validate_single_method_config(call, meth_config)
+
+    @classmethod
+    def _validate_single_method_config(cls, call, meth_config):
+        assert isinstance(
+            meth_config, MethodConfig
+        ), f"method_configs['{call}'] = has type {type(meth_config)} but only MethodConfig is allowed."
+        assert (
+            not meth_config.is_never()
+        ), f"method_configs['{call}'] specifies NEVER to be called; for driving the test, each configured method should ACTUALLY be tested."
+        assert (
+            meth_config.clear_state is CallConfig.NEVER
+        ), "meth_config's clear_state must be None, but isn't"
```

Comment:
nit: assert msg for `meth_config.clear_state != CallConfig.NEVER` should be polished a bit to something like: `unexpected value set to method_config's clear_state`, for it is enum over `CallConfig` rather than `None`

Reply:
see https://github.com/algorand/pyteal/pull/634/commits/5b98699bd363e8290219c57fff41942fb7ca60dc

Reply:
lgtm, resolving.